server/src/apub/fetcher.rs

```diff
@@ -45,11 +45,36 @@ pub enum SearchAcceptedObjects {
 /// Attempt to parse the query as URL, and fetch an ActivityPub object from it.
 ///
 /// Some working examples for use with the docker/federation/ setup:
-/// http://lemmy_alpha:8540/c/main
-/// http://lemmy_alpha:8540/u/lemmy_alpha
+/// http://lemmy_alpha:8540/c/main, or !main@lemmy_alpha:8540
+/// http://lemmy_alpha:8540/u/lemmy_alpha, or @lemmy_alpha@lemmy_alpha:8540
 /// http://lemmy_alpha:8540/p/3
 pub fn search_by_apub_id(query: &str, conn: &PgConnection) -> Result<SearchResponse, Error> {
-  let query_url = Url::parse(&query)?;
+  // Parse the shorthand query url
+  let query_url = if query.contains('@') {
+    debug!("{}", query);
+    let split = query.split('@').collect::<Vec<&str>>();
+
+    // User type will look like ['', username, instance]
+    // Community will look like [!community, instance]
+    let (name, instance) = if split.len() == 3 {
+      (format!("/u/{}", split[1]), split[2])
+    } else if split.len() == 2 {
+      if split[0].contains('!') {
+        let split2 = split[0].split('!').collect::<Vec<&str>>();
+        (format!("/c/{}", split2[1]), split[1])
```

Comment:
Maybe use replace() instead of split() here?

Reply:
Both work, but it doesn't seem like that would save anything. The only thing that's kinda frustrating to me, is that the iterator collect isn't a result, since it can easily panic if its out of bounds. I don't know how to do that tho.

Reply:
Maybe it would be better to replace the whole thing with a regex (or two). You can look at routes::webfinger for an example.